packages/react-components/src/hooks/useAlarms/transformers/parseAlarmStateProperty.ts

```diff
@@ -0,0 +1,44 @@
+import { AssetPropertyValue } from '@aws-sdk/client-iotsitewise';
+import { ALARM_STATUS } from '../constants';
+import { toTimestamp } from '../../../utils/time';
+
+export type UpperCaseStateName = keyof typeof ALARM_STATUS;
+export type PascalCaseStateName = (typeof ALARM_STATUS)[UpperCaseStateName];
+
+export const parseAlarmStateAssetProperty = (
+  assetPropertyValue?: AssetPropertyValue
+) => {
+  if (assetPropertyValue == null) {
+    return undefined;
+  }
+
+  const { timestamp, quality, value } = assetPropertyValue;
+
+  const alarmStateJSON = value?.stringValue;
+
+  if (alarmStateJSON == null) {
+    return undefined;
+  }
+
+  try {
+    const { stateName } = JSON.parse(alarmStateJSON);
```

Comment:
Let's use a type-safe JSON parse with a cast to an internal type that matches the expected state value

Reply:
Is there a library we are using for this? Otherwise I can make a safeParse function

Reply:
I don't know of any good external libraries, we should write one if it's not too much of a headache